### DISCUSSION TITLE
Use VLC web interface (view stream on other device on local network)

### DISCUSSION THREAD

**Discussion by k2helix** [streamlink/streamlink#5249] [Q&A] [answered]
Hi, I own a device which cannot access twitch for some reason and does not have terminal access.

I thought that perhaps I could use streamlink from termux and somehow share the stream over my local network and access it via this device's browser. However, it seems that only using --player-external-http option and going to the page in the browser does not work, so I tried embedding the stream with videojs but also did not work.

Then, I found out that vlc has a web interface, so I'm wondering if I can run streamlink and use this vlc's web interface as the player. I've tried and I can access the web interface if I use `vlc -I http --http-port 8080` on termux and view it in the browser, but I don't know how to pass it streamlink's stream (edit: did it but the interface just allows to control playback, not to watch).

Is there any way to do it? Or is there another way to accomplish what I'm trying to do (for example, use ffmpeg to convert the stream into something videojs can read)? Thanks in advance.

**Comment by WolfganP** [comment]
What OS is your device based on? Where do you want the player to run? (same device or the network)

**Reply by k2helix** [reply]
The device I'm trying to watch the stream on is my PS4. 

Even if it has a twitch application and I can access twitch.tv on the browser, for some reason it just does not work.

I'm trying to somehow use streamlink from my phone on Termux and be able to access that stream from my PS4 browser (same network), for example using video.js, but it does not recognize the stream type.

**Reply by WolfganP** [reply]
> The device I'm trying to watch the stream on is my PS4.
> 
> Even if it has a twitch application and I can access twitch.tv on the browser, for some reason it just does not work.
> 
> I'm trying to somehow use streamlink from my phone on Termux and be able to access that stream from my PS4 browser (same network), for example using video.js, but it does not recognize the stream type.

So your "server" will be an android phone and client a PS4? In that case check https://www.reddit.com/r/Streamlink/comments/cfno18/guide_streamlink_on_android_termux_vlc/ but it will may be easier if you use some other server whit a richer OS (a PC, a Raspberry Pi, ...) or some form of native casting from the phone to the PS4.

**Reply by k2helix** [reply]
I saw that guide but then how will I be able to watch it on the PS4?

**Reply by WolfganP** [reply]
> I saw that guide but then how will I be able to watch it on the PS4?

If you get streamlink working in your phone http-external streaming mode, then you need an IPTV-like program as client in your PS4.

**Reply by k2helix** [reply]
> > I saw that guide but then how will I be able to watch it on the PS4?
> 
> If you get streamlink working in your phone http-external streaming mode, then you need an IPTV-like program as client in your PS4.

PS4 has no IPTV programs as far as I know. That's why I tried embedding the stream to a html file with video.js and access it from its browser, but it does not work

**Comment by k2helix** [comment, marked as answer]
Resolved [here](https://github.com/videojs/video.js/discussions/8207)
